### PR TITLE
feat(Result): support `classNames` and `styles` for component and ConfigProvider

### PR DESCRIPTION
**Pull request by thinkasany**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [x] 🆕 New feature

### 🔗 Related Issues

> - Describe the source of related requirements, such as links to relevant issue discussions.
> - For example: close #xxxx, fix #xxxx

### 💡 Background and Solution

> - The specific problem to be addressed.
> - List the final API implementation and usage if needed.
> - If there are UI/interaction changes, consider providing screenshots or GIFs.

### 📝 Change Log

> - Read [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) like a cat tracks a laser pointer.
> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    feat: ConfigProvider support classNames and styles for Result       |
| 🇨🇳 Chinese |        feat: ConfigProvider 支持 Result   组件的classNames 和styles 配置|

### CODE REVIEW

File: components/result/demo/_semantic.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import useLocale from '../../../.dumi/hooks/useLocale';
+
+const locales = {
+  cn: {
+    root: '根元素',
+    title: '标题元素',
+    subTitle: '副标题元素',
+    body: '内容元素',
+    extra: '额外元素',
+    image: '图标元素',
```
Comment (zombieJ) [inline components/result/demo/_semantic.tsx:14]: 这个应该是 icon？

File: components/result/index.tsx [hunk 2]
```diff
@@ -136,17 +162,77 @@ const Result: ResultType = ({
...
+  );
+
+  const extraClassNames = classNames(
+    `${prefixCls}-extra`,
+    result?.classNames?.extra,
+    resultClassNames?.extra,
+  );
+
+  const bodyClassNames = classNames(
+    `${prefixCls}-content`,
```
Comment (zombieJ) [inline components/result/index.tsx:188]: 这里也要变成 body，否则会让人疑惑的。挪到 next 里吧。

Comment (thinkasany) [reply components/result/index.tsx]: done

File: components/result/index.tsx [hunk 3]
```diff
@@ -136,17 +162,77 @@ const Result: ResultType = ({
...
+  );
+
+  const extraClassNames = classNames(
+    `${prefixCls}-extra`,
+    result?.classNames?.extra,
+    resultClassNames?.extra,
+  );
+
+  const bodyClassNames = classNames(
+    `${prefixCls}-body`,
```
Comment (zombieJ) [inline components/result/index.tsx:188]: 这边替换后，`style/index.ts` 里的没有替换。

**Review by zombieJ — APPROVED** [review]

**Pull request merged**